src/backend/joanie/core/models/courses.py

```diff
@@ -168,6 +168,40 @@ def clean(self):
         self.code = utils.normalize_code(self.code)
         return super().clean()
 
+    def get_abilities(self, user=None, auth=None):
+        """
+        Compute and return abilities for a given user taking into account
+        the current state of the object.
+        """
+        is_owner_or_admin = False
+
+        # Avoid hitting the database by checking authentication via the auth token first
+        username = (
+            auth["username"]
+            if auth
+            else (user.username if user and user.is_authenticated else None)
+        )
+        if username:
+            try:
+                role = self.user_role
+            except AttributeError:
+                try:
+                    role = self.accesses.filter(user__username=username).values("role")[
+                        0
+                    ]["role"]
+                except (OrganizationAccess.DoesNotExist, IndexError):
+                    role = None
+
+            is_owner_or_admin = role in [enums.OWNER, enums.ADMIN]
+
+        return {
+            "get": True,
+            "patch": is_owner_or_admin,
+            "put": is_owner_or_admin,
+            "delete": is_owner_or_admin,
```

Comment:
delete an organization should be only allowed by an owner WDYT ?

Reply:
Oh yeah true!

Reply:
same for course